src/indexing.jl

```diff
@@ -1,8 +1,17 @@
 
-@inline function getindex(x::SMatrix{4, 4}, i::T, j::T) where T <: Union{DNA, RNA}
-    return x[trailing_zeros(i) + 1, trailing_zeros(j) + 1]
+@inline function Base.checkbounds(x::SMatrix{4, 4}, i::T, j::T) where T <: NucleicAcid
```

Comment:
Add checkbounds to make sure uncertain nucleotides like 'M' are caught. This can be over-ridden then when indexing into an SMatrix using @inbounds.